lib/spack/spack/ci.py

```diff
@@ -1029,6 +1029,7 @@ def main_script_replacements(cmd):
             job_vars = job_object.setdefault("variables", {})
             job_vars["SPACK_JOB_SPEC_DAG_HASH"] = release_spec_dag_hash
             job_vars["SPACK_JOB_SPEC_PKG_NAME"] = release_spec.name
+            job_vars["SPACK_JOB_SHORT_SPEC"] = release_spec.short_spec
```

Comment:
```suggestion
            id_format  = "{name}{@version}{%compiler.name}{@compiler.version}{variants}{arch=architecture}"
            
            job_vars = job_object.setdefault("variables", {})
            job_vars["SPACK_JOB_SPEC_DAG_HASH"] = release_spec_dag_hash
            job_vars["SPACK_JOB_SPEC_PKG_NAME"] = release_spec.name
            job_vars["SPACK_JOB_SHORT_SPEC"] = release_spec.format(id_format)
```

I'd probably vote for something like this. (Although format without args will output the same. I'd be a little worried that might change at some point and so good to enforce a format for ID-ing an abstract spec.)